### PR TITLE
MODQM-154: MARC Authority - Leader rules

### PR DESCRIPTION
**Pull request by Shans-Kaluhin**
### Purpose
Marc authority leader rules

### Approach
Add leader validator for authority

### Learning
[MODQM-154](https://issues.folio.org/browse/MODQM-154)

### CODE REVIEW

File: src/test/java/org/folio/qm/validation/leader/AuthorityLeaderValidationRuleTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+class AuthorityLeaderValidationRuleTest {
+
+  private LeaderValidationRule rule = new AuthorityLeaderValidationRule();
+  public static final String VALID_LEADER = "00241xz\\\\a2200109nu\\4500";
+  public static final String WRONG_RECORD_STATUS = "00241zz\\\\a2200109nu\\4500";
+  public static final String WRONG_RECORD_TYPE = "00241xx\\\\a2200109nu\\4500";
+  public static final String WRONG_UNDEFINED_CHARACTER = "00241xzz\\a2200109nu\\4500";
+  public static final String WRONG_ENCODING_LEVEL = "00241xz\\\\a2200109au\\4500";
+  public static final String WRONG_PUNCTUATION_POLICY = "00241xz\\\\a2200109nz\\4500";
```
Comment (psmagin) [inline src/test/java/org/folio/qm/validation/leader/AuthorityLeaderValidationRuleTest.java:24]: This could be private

Comment (oleksandr-dekin) [reply src/test/java/org/folio/qm/validation/leader/AuthorityLeaderValidationRuleTest.java:24]: in this case, please, marc as private fields for these class too: `BibliographicLeaderValidationRuleTest`, `CommonLeaderValidationRuleTest`, `HoldingsLeaderValidationRuleTest`

**Review by oleksandr-dekin — APPROVED** [review]

File: src/main/resources/swagger.api/schemas/marcFormat.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
   "additionalProperties": false,
   "enum": [
     "BIBLIOGRAPHIC",
-    "HOLDINGS"
+    "HOLDINGS",
+    "AUTHORITY"
```
Comment (psmagin) [inline src/main/resources/swagger.api/schemas/marcFormat.json:10]: You should bump minor interface version in case of changing schemas

**Review by psmagin — CHANGES_REQUESTED** [review]

File: src/main/resources/swagger.api/schemas/marcFormat.json (hunk 2 above)
Comment (Shans-Kaluhin) [reply src/main/resources/swagger.api/schemas/marcFormat.json:10]: Upgrade `records-editor.yaml` and `records-editor-async.yaml` versions to 1.1.1 or 1.2.0?

**Review by psmagin — APPROVED** [review]

**Pull request merged**